### PR TITLE
Refactor examples and modules with run command flag change

### LINKED ISSUE #18709: Refactor examples and modules with run command flag change
**Description:**
Refactor documentation based on https://github.com/ballerina-platform/ballerina-lang/pull/18633 change

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Required examples and documentation changes for https://github.com/ballerina-platform/ballerina-lang/pull/18633

Fixes #18709

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/config-api/src/main/ballerina/src/config/Module.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 The `ballerina/config` module provides the Config API to read configurations from environment variables, TOML files, and command-line parameters, and build a consolidated set of configurations.
```
Comment (praneesha) [inline stdlib/config-api/src/main/ballerina/src/config/Module.md]:
```suggestion
The `ballerina/config` module provides the Config API to read configurations from environment variables, TOML files, and command-line parameters and build a consolidated set of configurations.
```

File: stdlib/config-api/src/main/ballerina/src/config/Module.md [hunk 2]
```diff
@@ -16,10 +16,12 @@ The Config API provides the capability to feed sensitive data (e.g., passwords)
 
 ### Setting configurations
 
-To explicitly specify a configuration file, the `--config` or `-c` flag can be used. If this flag is not set when running a project, Ballerina looks for a `ballerina.conf` file in project root. When running a single file or a `.balx`, it's picked from the same directory as the `.balx` or source. The path to the configuration file can either be an absolute or a relative path. 
+To explicitly specify a configuration file, the `--b7a.config=<path to configuration file>` property can be used. If
```
Comment (praneesha) [inline stdlib/config-api/src/main/ballerina/src/config/Module.md]:
```suggestion
To specify a configuration file explicitly, the `--b7a.config=<path to configuration file>` property can be used. If 
```

File: stdlib/config-api/src/main/ballerina/src/config/Module.md [hunk 3]
```diff
@@ -16,10 +16,12 @@ The Config API provides the capability to feed sensitive data (e.g., passwords)
 
 ### Setting configurations
 
-To explicitly specify a configuration file, the `--config` or `-c` flag can be used. If this flag is not set when running a project, Ballerina looks for a `ballerina.conf` file in project root. When running a single file or a `.balx`, it's picked from the same directory as the `.balx` or source. The path to the configuration file can either be an absolute or a relative path. 
+To explicitly specify a configuration file, the `--b7a.config=<path to configuration file>` property can be used. If 
+this property is not set when 
+running a project, Ballerina looks for a `ballerina.conf` file in project root. When running a single file or a `.balx`, it's picked from the same directory as the `.balx` or source. The path to the configuration file can either be an absolute or a relative path.
```
Comment (praneesha) [inline stdlib/config-api/src/main/ballerina/src/config/Module.md]:
```suggestion
running a project, Ballerina looks for a `ballerina.conf` file in the project root. When running a single file or a `.balx`, it's picked from the same directory in which the `.balx` or source resides. The path to the configuration file can either be an absolute or a relative path. 
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help [hunk 4]
```diff
@@ -54,23 +46,19 @@ OPTIONS
...
-        Provides Ballerina VM options to be used by the runtime
-        when executing a Ballerina program or module. VM options
-        are key/value pairs. If multiple VM options need to be
-        provided, each option has to be specified using its
-        own -B flag.
+        
+    configs(--key=value)...
+        Sets Ballerina environment parameters as key/value pairs.
+        If multiple parameters need to be provided, each parameter
+        has to be specified using its own -e flag. Details of the
```
Comment (pubudu91) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: This contains a reference to the `-e` flag.

Comment (pubudu91) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]:
```suggestion
        has to be preceded by '--'. Details of the
```

Comment (warunalakshitha) [reply cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: Fixed as suggested.

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help [hunk 5]
```diff
@@ -54,23 +46,19 @@ OPTIONS
...
+        environment parameters that the Ballerina standard library
+        modules support can be found in the relevant API
+        documentation.
+	        
+    --
+        The optional argument ‘--’ terminates all configs. Any following
```
Comment (pubudu91) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: How about something like the below for this section
```
The argument ‘--’ signals the end of Ballerina environment parameters. Any arguments following '--' are treated as arguments to the program even if they start with '--'.

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help [hunk 6]
```diff
@@ -54,23 +46,19 @@ OPTIONS
...
+        If multiple parameters need to be provided, each parameter
+        has to be specified using its own -e flag. Details of the
+        environment parameters that the Ballerina standard library
+        modules support can be found in the relevant API
+        documentation.
+	        
+    --
+        The optional argument ‘--’ terminates all configs. Any following 
+        arguments are treated as args to program even they are start with '--'. 
+        This is required only if specific arg needs to start with '--'.
```
Comment (pubudu91) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: what does this mean?

File: examples/config-api/config_api.out [hunk 7]
```diff
@@ -20,23 +20,23 @@ $ ballerina run config_api.bal
 ballerina: enter secret for config value decryption:
 
 
-# To explicitly specify a configuration file, use either the `--config` or the `-c` flag.
-# If this flag is not set, Ballerina looks for a `ballerina.conf` file in the directory in which the source files
+# To explicitly specify a configuration file, use `--b7a.config=<path to configuration file>` property.
```
Comment (pubudu91) [inline examples/config-api/config_api.out]: Is it possible to change this property to `b7a.config.file` or something? The convention we have used for params is `<module-name>.<whatever-param-name>`.

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help (hunk 6 above)
Comment (warunalakshitha) [reply cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: What I mean was you don't need to always add '--' before args and say this is optional

File: examples/config-api/config_api.out (hunk 7 above)
Comment (warunalakshitha) [reply examples/config-api/config_api.out]: Initial i added like b7a.config.file.path and later change that to b7a.config since bit shorter.

Comment (warunalakshitha) [reply examples/config-api/config_api.out]: Lets change it to b7a.config.file as per convention.

File: examples/http-caching-client/http_caching_client.server.out [hunk 8]
```diff
@@ -1,8 +1,8 @@
 # The two services have to be run separately to observe the following output. 
 # For clarity, only the relevant parts of the HTTP trace logs have been included here.
 # At the command line, navigate to the directory that contains the
-# `.bal` file and run the `ballerina run` command by setting `-e b7a.http.tracelog.console=true` to start the service.
-$ ballerina run caching_proxy.bal -e b7a.http.tracelog.console=true
+# `.bal` file and run the `ballerina run` command by setting `--b7a.http.tracelog.console=true` to start the service.
+$ ballerina run --b7a.http.tracelog.console=true caching_proxy.bal
```
Comment (pubudu91) [inline examples/http-caching-client/http_caching_client.server.out]:
```suggestion
$ ballerina run caching_proxy.bal --b7a.http.tracelog.console=true
```

File: stdlib/config-api/src/main/ballerina/src/config/Module.md [hunk 9]
```diff
@@ -3,7 +3,7 @@
 The `ballerina/config` module provides the Config API to read configurations from environment variables, TOML files, and command-line parameters, and build a consolidated set of configurations.
 
 The precedence order for configuration lookup is as follows: 
-1. CLI parameters (used with the -e flag)
+1. CLI configurations (used with the --)
```
Comment (pubudu91) [inline stdlib/config-api/src/main/ballerina/src/config/Module.md]: This should actually be CLI arguments right? CLI configurations sound like it's configs for the CLI

File: stdlib/log-api/src/main/ballerina/src/log/Module.md [hunk 10]
```diff
@@ -29,7 +29,7 @@ b7a.log.level="<LOG_LEVEL>"
 
 The log level can also be configured through a CLI parameter as follows:
 ```bash
-$ ballerina run program.bal -e b7a.log.level=<LOG_LEVEL>
+$ ballerina run --b7a.log.level=<LOG_LEVEL> program.bal
```
Comment (pubudu91) [inline stdlib/log-api/src/main/ballerina/src/log/Module.md]:
```suggestion
$ ballerina run program.bal --b7a.log.level=<LOG_LEVEL>
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help [hunk 11]
```diff
@@ -54,23 +46,19 @@ OPTIONS
...
+        Sets Ballerina environment parameters as key/value pairs.
+        If multiple parameters need to be provided, each parameter
+        has to be preceded by '--'. Details of the
+        environment parameters that the Ballerina standard library
+        modules support can be found in the relevant API
+        documentation.
+	        
+    --
+        The optional argument ‘--’ terminates all configs. Any succeeding 
+        arguments are treated as args of the program even if they start with '--'.
```
Comment (pubudu91) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]:
```suggestion
        arguments are treated as arguments of the program even if they start with '--'. 
```
Better to refrain from using abbreviations etc. such as "args".

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help [hunk 12]
```diff
@@ -54,23 +46,19 @@ OPTIONS
...
+        If multiple parameters need to be provided, each parameter
+        has to be preceded by '--'. Details of the
+        environment parameters that the Ballerina standard library
+        modules support can be found in the relevant API
+        documentation.
+	        
+    --
+        The optional argument ‘--’ terminates all configs. Any succeeding 
+        arguments are treated as args of the program even if they start with '--'. 
+        This is required only if a specific arg needs to start with '--'.
```
Comment (pubudu91) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: If I understand correctly, what this means is the `--` is required if the user wants to give an argument starting with `--` to his program right?

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help (hunk 5 above)
Comment (warunalakshitha) [reply cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: Fixed

File: stdlib/config-api/src/main/ballerina/src/config/Module.md (hunk 9 above)
Comment (warunalakshitha) [reply stdlib/config-api/src/main/ballerina/src/config/Module.md]: +1 Replaced configurations with arguments.

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help (hunk 12 above)
Comment (warunalakshitha) [reply cli/ballerina-tool/src/main/resources/cli-help/ballerina-run.help]: Yes. I change the description as suggested and seems fine.

**Pull request merged**